stl/src/fiopen.cpp

```diff
@@ -44,6 +44,10 @@ FILE* _Xfiopen(const CharT* filename, ios_base::openmode mode, int prot) {
         0,
     };
 
+    if (mode == 0) {
+        return nullptr; // no valid mode
+    }
+

```

Comment:
I think this part is unnecessary. It is odd to optimize _performance_ of negative case by making positive case less optimal. And I don't think a function that is supposed to open a file needs fine instruction-level optimization.

I would drop this part of change, and keep the rest as an optimization of _readability_.

Reply:
Positive case is actually NOT less optimal.

Negative case ALWAYS has to be checked first because otherwise it will cause a bug if mode == 0

Reply:
I took this into serious consideration with this patch. Positive case is the same. Optimizing for the negative case without affecting the positive is a good thing

Reply:
The original code also checks for the negative first.